src/wasm/types.rs

```diff
@@ -119,6 +119,13 @@ type CardanoCertificate =
         keypath: Keypath
         poolKeyhash: Uint8Array
       }
+    }
+  | {
+      voteDelegation: {
+        keypath: Keypath
+        type: number
```

Comment:
`number` is a bit too generic, wouldn't it be possible to narrow the values down with some enum? That would potentially allow for making this a union of different object shapes, making it clear when is drep hash required and when it isn't

Moreover, if possible, instead of null, I'd consider making the `drepCredHash` optional, at least from a typescript perspective it would feel more "natural"

Reply:
I agree, this type should be something like 

```
CardanoDRepType = 'keyHash'  | 'scriptHash' | 'alwaysAbstain'  | 'alwaysNoConfidence'
```

if it doesn't work out of the box like this, check other enums (e.g. in btc.proto/btc.rs) and scripts/build-protos.rs for reference,